python/lsst/daf/butler/core/_butlerUri/s3.py

```diff
@@ -39,6 +41,9 @@
 from ._butlerUri import ButlerURI
 from .s3utils import getS3Client, s3CheckFileExists, bucketExists
 
+from botocore.exceptions import ClientError
+from http.client import ImproperConnectionState, HTTPException
+from urllib3.exceptions import RequestError, HTTPError
```

Comment:
Since these are only used once, it may be better to import the module and use a fully-qualified name instead.  On the other hand, that makes everything in the module available (unnecessarily) as opposed to just the imported symbol(s).

Reply:
Is there any difference to python (performance or otherwise) to plucking only needed symbols from a module?

Reply:
I don't think so. It's more of a convention where you don't want ambiguous bare symbols turning up in the code. In this particular case I think `HTTPError` is fairly unambiguous where it came from so I'm fine with leaving it as is.

Reply:
Well, if it were obvious, the comments wouldn't be needed at point-of-use :).  But I'm also fine with leaving it.